cranelift/codegen/src/isa/x64/pcc.rs

```diff
@@ -0,0 +1,600 @@
+            src: XmmMemAligned(RegMem::Mem { ref addr }),
+            ..
+        }
+        | Inst::XmmUnaryRmREvex {
+            src: XmmMem(RegMem::Mem { ref addr }),
+            ..
+        }
+        | Inst::XmmUnaryRmRVex {
+            src: XmmMem(RegMem::Mem { ref addr }),
+            ..
+        }
+        | Inst::XmmUnaryRmRImmVex {
+            src: XmmMem(RegMem::Mem { ref addr }),
+            ..
+        }
+        | Inst::XmmRmREvex {
+            src2: XmmMem(RegMem::Mem { ref addr }),
+            ..
+        }
+        | Inst::XmmUnaryRmRImmEvex {
+            src: XmmMem(RegMem::Mem { ref addr }),
+            ..
+        }
+        | Inst::XmmRmREvex3 {
+            src3: XmmMem(RegMem::Mem { ref addr }),
+            ..
+        } => {
+            check_load(ctx, None, addr, vcode, I8X16, 128)?;
+            Ok(())
+        }
+
+        Inst::XmmVexPinsr {
+            src2: GprMem(RegMem::Mem { ref addr }),
+            ..
+        }
+        | Inst::GprToXmm {
+            src: GprMem(RegMem::Mem { ref addr }),
+            ..
+        }
+        | Inst::GprToXmmVex {
+            src: GprMem(RegMem::Mem { ref addr }),
+            ..
+        } => {
+            check_load(ctx, None, addr, vcode, I64, 64)?;
+            Ok(())
+        }
+
+        Inst::XmmCmove {
+            consequent: XmmMemAligned(RegMem::Mem { ref addr }),
+            ..
+        } => {
+            check_load(ctx, None, addr, vcode, I8X16, 128)?;
+            Ok(())
+        }
+
+        Inst::XmmCmpRmR {
+            src: XmmMemAligned(RegMem::Mem { ref addr }),
+            ..
+        }
+        | Inst::XmmRmRImm {
+            src2: RegMem::Mem { ref addr },
+            ..
+        } => {
+            check_load(ctx, None, addr, vcode, I8X16, 128)?;
+            Ok(())
+        }
+
+        Inst::CvtIntToFloat {
+            src2: GprMem(RegMem::Mem { ref addr }),
+            ..
+        }
+        | Inst::CvtIntToFloatVex {
+            src2: GprMem(RegMem::Mem { ref addr }),
+            ..
+        } => {
+            check_load(ctx, None, addr, vcode, I64, 64)?;
+            Ok(())
+        }
+
+        Inst::LockCmpxchg { mem: ref dst, .. } | Inst::AtomicRmwSeq { mem: ref dst, .. } => {
+            check_store(ctx, None, dst, vcode, I64)?;
+            Ok(())
+        }
+
+        Inst::CallUnknown {
+            dest: RegMem::Mem { ref addr },
+            ..
+        }
+        | Inst::ReturnCallUnknown {
+            callee: RegMem::Mem { ref addr },
+            ..
+        }
+        | Inst::JmpUnknown {
+            target: RegMem::Mem { ref addr },
+            ..
+        } => {
+            check_load(ctx, None, addr, vcode, I64, 64)?;
+            Ok(())
+        }
+
+        _ if vcode.inst_defines_facts(inst_idx) => {
+            trace!(
+                "Unsupported inst during PCC validation: {:?}",
+                vcode[inst_idx]
+            );
+            Err(PccError::UnsupportedFact)
```

Comment:
It seems like the trace here should be the `Display`/`Debug` of the error returned, and the error should contain the instruction or instruction index.

That is a bit larger of a refactor, and can happen in a follow up PR, but we should at least in the meantime make this a `log::error!` rather than a `crate::trace!`.

Reply:
Added one in the toplevel driver loop, thanks!